File: snapdrama/apps/api/app/schemas/drama.py
from pydantic import BaseModel

class DramaBase(BaseModel):
    title: str
    description: str | None = None
    thumbnail_url: str | None = None
    genre_id: str
    release_year: int
    total_episodes: int

class DramaCreate(DramaBase):
    pass

class DramaUpdate(DramaBase):
    title: str | None = None
    genre_id: str | None = None
    release_year: int | None = None
    total_episodes: int | None = None

class DramaInDB(DramaBase):
    id: str

    class Config:
        from_attributes = True

class Drama(DramaInDB):
    pass 

File: snapdrama/packages/shared/lib/src/models/drama.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'drama.freezed.dart';
part 'drama.g.dart';

@freezed
class Drama with _$Drama {
  const factory Drama({
    required String id,
    required String title,
    required String description,
    required String thumbnailUrl,
    required String genreId,
    required int releaseYear,
    required int totalEpisodes,
  }) = _Drama;

  factory Drama.fromJson(Map<String, dynamic> json) => _$DramaFromJson(json);

  @override
  // TODO: implement description
  String get description => throw UnimplementedError();

  @override
  // TODO: implement genreId
  String get genreId => throw UnimplementedError();

  @override
  // TODO: implement id
  String get id => throw UnimplementedError();

  @override
  // TODO: implement releaseYear
  int get releaseYear => throw UnimplementedError();

  @override
  // TODO: implement thumbnailUrl
  String get thumbnailUrl => throw UnimplementedError();

  @override
  // TODO: implement title
  String get title => throw UnimplementedError();

  @override
  Map<String, dynamic> toJson() {
    // TODO: implement toJson
    throw UnimplementedError();
  }

  @override
  // TODO: implement totalEpisodes
  int get totalEpisodes => throw UnimplementedError();
}
